Dockerfile

```diff
@@ -45,13 +46,23 @@ RUN dnf --nodocs --setopt install_weak_deps=false -y install \
     && dnf --nodocs --setopt install_weak_deps=false -y upgrade --security \
     && dnf clean all
 
-# Before copying the entire source, copy just requirements.txt.
-# This makes podman cache this (lengthy) step as long as requirements.txt stays unchanged.
-# Without this, any change in src/ would make pip install run again.
-COPY ./requirements.txt /opt/app-root/src/requirements.txt
-RUN pip3 install --no-deps -r /opt/app-root/src/requirements.txt && \
-    rm -f /opt/app-root/src/requirements.txt
+# Install uv
+COPY --from=ghcr.io/astral-sh/uv:latest /uv /uvx /bin/
+
+# Before copying the entire source, copy just the dependency files
+# This makes podman cache this (lengthy) step as long as the dependency files stays unchanged.
+# Without this, any change in src/ would make uv sync
+COPY ./pyproject.toml ./uv.lock /opt/app-root/src/
+
+# Sync project dependencies into a virtual environment 
```

Comment:
Is there a way for uv to install into the system's python package store? In an image we don't _really_ need a venv